### PR TITLE
s9y-db importer permalink, tag, excerpt, and semantic HTML improvements

### PR DESCRIPTION
**Pull request by judebert**
These commits support plugins and properties required to migrate judebert.com from S9Y to Jekyll. The features include:
* support permalinks from the `entryproperties` plugin
* duplicate S9Y "extended body" functionality with Jekyll excerpts
* import tags and categories
* convert iframes and Serendipity image/captions to semantic HTML figures
* use relative links instead of absolute
* entryincludes plugin

Documentation has been updated to describe the new functionality.

### CODE REVIEW

File: lib/jekyll-import/importers/s9y_database.rb [hunk 1]
```diff
@@ -56,37 +62,70 @@ def self.specify_options(c)
...
+          :excerpt_separator => opts.fetch("excerpt_separator", "<a id=\"extended\"></a>"),
+          :includeentry      => opts.fetch("includeentry", false),
+          :imgfig            => opts.fetch("imgfig", true),
+          :linebreak         => opts.fetch("linebreak", "wp"),
+          :relative          => opts.fetch("relative", nil),
         }
 
-        options[:clean_entities] = require_if_available("htmlentities", "clean_entities") if options[:clean_entities]
+        if options[:clean_entities]
+          options[:clean_entities] = require_if_available("htmlentities", "clean_entities") if options[:clean_entities]
```
[the remark this answers is not in the document]
Comment (judebert) [reply lib/jekyll-import/importers/s9y_database.rb]: This appears to be a merge error. My humblest apologies. Thanks for catching this!

**Comment by judebert** [comment]
All PR comments addressed. script/cibuild passes on my computer.

**Comment by ashmaroli** [comment]
I still the following in the diff:
```ruby
if options[:clean_entities]
  options[:clean_entities] = require_if_available("htmlentities", "clean_entities") if options[:clean_entities]
end

if options[:markdown]
  options[:markdown] = require_if_available("reverse_markdown", "markdown") if options[:markdown]
end
```
And I see RuboCop errors in our Travis logs. So you may not have an up-to-date RuboCop locally.
Please run the following:
```
bundle update
bundle exec rubocop -a
```

File: docs/_importers/s9ydatabase.md [hunk 2]
```diff
@@ -45,3 +45,80 @@ $ ruby -r rubygems -e 'require "jekyll-import";
...
+        <code>http://myhost.com/blog/lifestyle/7-lucky-post.html</code> will be
+        converted to <code>/lifestyle/7-lucky-post.html</code>. Note that you
+        should not include the trailing <code>/</code> slash, and that https:
+        is not yet supported.
+      </p>
+    </li><li>
+      <strong>linebreak</strong>
+      <p>
+        If you used a formatting extension, this options might be useful. Use
+        one of these options to try and replicate your post line breaks as
```
[the remark this answers is not in the document]
Comment (judebert) [reply docs/_importers/s9ydatabase.md]: I'm not sure I understand. Could you provide an example to clarify?

File: docs/_importers/s9ydatabase.md [hunk 3]
```diff
@@ -45,3 +45,80 @@ $ ruby -r rubygems -e 'require "jekyll-import";
...
+        closely as possible:
+      </p>
+        <li><strong>wp<strong> (the default)</li>
+        <p>
+          Replicate the Wordpress linebreak behavior, the default for S9Y.
```
[the remark this answers is not in the document]
Comment (judebert) [reply docs/_importers/s9ydatabase.md]: Will fix.

File: docs/_importers/s9ydatabase.md [hunk 4]
```diff
@@ -45,3 +45,80 @@ $ ruby -r rubygems -e 'require "jekyll-import";
...
+        </p>
+        <li><strong>nokogiri</strong></li>
+        <p>
+          Uses the <code>nokogiri</code> gem to interpret entries as XHTML
+          formatted.  If you write HTML entries, this preserves the HTML
+          linebreaks.
```
[the remark this answers is not in the document]
Comment (judebert) [reply docs/_importers/s9ydatabase.md]: Will fix

File: docs/_importers/s9ydatabase.md [hunk 5]
```diff
@@ -45,3 +45,80 @@ $ ruby -r rubygems -e 'require "jekyll-import";
...
+  <p>
+    This importer now supports some of the most common S9Y plugins.
+  </p>
+  <ul>
+    <li>
+      <strong>includeentry</strong>
+      <p>
+        Set this true to transclude entries like the <code>includeentry</code>
+        plugin. The current, static content of the entry will be included.
+        Future updates will not be synched.
```
[the remark this answers is not in the document]
Comment (judebert) [reply docs/_importers/s9ydatabase.md]: Will fix

File: docs/_importers/s9ydatabase.md [hunk 6]
```diff
@@ -45,3 +45,80 @@ $ ruby -r rubygems -e 'require "jekyll-import";
...
+        the post as an excerpt. This option allows you to restore the S9Y
+        behavior: posts with extended body will have the specified separator
+        added to their front matter, and you can modify your index layout to
+        show the excerpts.
+      </p>
+    </li><li>
+      <strong>imgfig</strong>
+      <p>
+        By default, this converts references to S9Y's media library into HTML
+        <code>figure</code> tags. Set it false to keep the original
```
[the remark this answers is not in the document]
Comment (judebert) [reply docs/_importers/s9ydatabase.md]: Will fix

File: docs/_importers/s9ydatabase.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashmaroli) [reply docs/_importers/s9ydatabase.md]:
```html
<p>
  If you used a formatting extension, this option might be useful. Use
  one of the following sub-options to try and replicate your post line
  breaks as closely as possible:

  <ul>
    <li>wp</li>
    <li>nokogiri</li>
  <ul/>
</p>
```

**Comment by judebert** [comment]
Friendly ping: I think I pushed another update addressing all issues, about 3 weeks ago. Is there anything else I need to do to merge?

**Comment by ashmaroli** [comment]
> addressing all issues, about 3 weeks ago.

Apologies, @judebert.
This had slipped out of my head. Will post a fresh review shortly.

**Review by parkr — APPROVED** [review]

**Comment by parkr** [comment]
@jekyllbot: merge +minor

**Pull request merged**